eng/pipelines/typespec-validation-all.yml

```diff
@@ -4,7 +4,19 @@ trigger:
     - main
     - typespec-next
 
-pr: none
+pr:
```

Comment:
Isn't the other pipeline already covering this check? I thought it would run on everything if some of the core files were mutated. Did we change that logic?

Reply:
The PR pipeline does validate all specs if "common" files are changed, using this code:

https://github.com/Azure/azure-rest-api-specs/blob/c06b7c22971d4c71e52f0962728c99aabf70127e/eng/scripts/Get-TypeSpec-Folders.ps1#L32-L44

However, one gap still remains, that the "All" pipeline runs on both Windows and Linux, while the "PR" pipeline only runs on Linux.

I recently merged a PR which unintentionally broke Windows:

Break: https://github.com/Azure/azure-rest-api-specs/pull/25466
Fix: https://github.com/Azure/azure-rest-api-specs/pull/25483

To prevent this from happening again, I think we should test all "common" changes on both Linux and Windows.  The easiest way is to add these triggers to the "All" pipeline.